packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -1669,6 +1694,155 @@ - (void)keyUp:(NSEvent *)event {
+  NSArray *fileNames = [pasteboard propertyListForType:NSFilenamesPboardType] ?: @[];
+  for (NSString *file in fileNames) {
+    NSURL *fileURL = [NSURL fileURLWithPath:file];
+    BOOL isDir = NO;
+    BOOL isValid = [[NSFileManager defaultManager] fileExistsAtPath:fileURL.path isDirectory:&isDir] && !isDir;
+    if (isValid) {
+      
+      NSString *MIMETypeString = nil;
+      if (fileURL.pathExtension) {
+        CFStringRef fileExtension = (__bridge CFStringRef)fileURL.pathExtension;
+        CFStringRef UTI = UTTypeCreatePreferredIdentifierForTag(kUTTagClassFilenameExtension, fileExtension, NULL);
+        if (UTI != NULL) {
+          CFStringRef MIMEType = UTTypeCopyPreferredTagWithClass(UTI, kUTTagClassMIMEType);
+          CFRelease(UTI);
+          MIMETypeString = (__bridge_transfer NSString *)MIMEType;
+        }
+      }
+
+      NSNumber *fileSizeValue = nil;
+      NSError *fileSizeError = nil;
+      BOOL success = [fileURL getResourceValue:&fileSizeValue
+                                        forKey:NSURLFileSizeKey
+                                         error:&fileSizeError];
+
+      DataTransferItem transferItem = {
+        .name = fileURL.lastPathComponent ? fileURL.lastPathComponent.UTF8String : "",
+        .kind = "file",
+        .type = MIMETypeString ? MIMETypeString.UTF8String : "",
+        .uri = fileURL.path ? fileURL.path.UTF8String : "",
+      };
+      
+      if (success) {
+        transferItem.size = fileSizeValue.intValue;
+      }
+
+      if ([MIMETypeString hasPrefix:@"image/"]) {
+        NSImage *image = [[NSImage alloc] initWithContentsOfURL:fileURL];
+        CGImageRef cgImage = [image CGImageForProposedRect:nil context:nil hints:nil];
+        transferItem.width = static_cast<int>(CGImageGetWidth(cgImage));
+        transferItem.height = static_cast<int>(CGImageGetHeight(cgImage));
+      }
+      
+      dataTransferItems.push_back(transferItem);
+    }
+  }
+  
+  NSPasteboardType imageType = [pasteboard availableTypeFromArray:@[NSPasteboardTypePNG, NSPasteboardTypeTIFF]];
+  if (imageType && fileNames.count == 0) {
+    NSString *MIMETypeString = imageType == NSPasteboardTypePNG ? @"image/png" : @"image/tiff";
+    NSData *imageData = [pasteboard dataForType:imageType];
+    NSImage *image = [[NSImage alloc] initWithData:imageData];
+    CGImageRef cgImage = [image CGImageForProposedRect:nil context:nil hints:nil];
+    
+    NSString *dataURLString = RCTDataURL(MIMETypeString, imageData).absoluteString;
+    
+    DataTransferItem transferItem = {
+      .kind = "image",
+      .type = MIMETypeString ? MIMETypeString.UTF8String : "",
+      .uri = dataURLString ? dataURLString.UTF8String : "",
+      .size = static_cast<int>(imageData.length),
+      .width = static_cast<int>(CGImageGetWidth(cgImage)),
+      .height = static_cast<int>(CGImageGetHeight(cgImage)),
```

Comment:
Should these fields be defined with `size_t` so we don't have to downcast and potentially truncating the number?

Reply:
Not sure what effect this has on the event sent to JS but sure

Reply:
Actually, it seems I cannot. 
<img width="633" height="118" alt="image" src="https://github.com/user-attachments/assets/77f56261-b1ff-4fb3-96aa-5e59b870a7a9" />

Reply:
Comparing against `PointerEvent.h`, I only see `int` used there too.